src/main/webapp/MarkerInfoWindow.js

```diff
@@ -19,52 +19,174 @@ class MarkerInfoWindow {
+      btn.id = MarkerInfoWindow.DELETE_ID;
+      return btn;
     }
 
-    /** Displays the info window on the map */
-    open() {
-      this.googleInfoWindow.open(myMap.googleMap, this.myMarker.googleMarker);
+    /** Returns the edit button for this marker*/
+    makeEditButton() {
+      let btn = document.createElement("button");
+      btn.id = MarkerInfoWindow.EDIT_ID;
+      return btn;
+    }
+
+    /** Puts together info window buttons */
+    makeLeftColumn() {
+      let leftCol = document.createElement("div");
+      leftCol.classList.add(MarkerInfoWindow.LEFT_COLUMN);
+      leftCol.appendChild(this.makeDeleteButton());
+      leftCol.appendChild(this.makeEditButton());
+      return leftCol;
+    }
+
+    /** Passes the form input to the marker and refreshes the info window*/
+    updateContent() {
+      let inputTitle = document.getElementById(MarkerInfoWindow.TITLE_INPUT);
+      let inputBody = document.getElementById(MarkerInfoWindow.BODY_INPUT);
+
+      if (inputTitle  && inputBody) {
+          this.myMarker.setContent(inputTitle.value, inputBody.value);
+          this.myMarker.editing = false;
+          inputTitle.value = "";
+          inputBody.value = "";
+      }
+      this.open(this.myMarker);
+    }
+
+    /** Builds the HTML for the input form for the marker description*/
+    makeContentForm() {
+      let form = document.createElement("div");
+
+      let titleInput = document.createElement("textarea");
+      titleInput.id = MarkerInfoWindow.TITLE_INPUT;
+      titleInput.placeholder = MarkerInfoWindow.DEFAULT_TITLE;
+
+      let bodyInput = document.createElement("textarea");
+      bodyInput.id = MarkerInfoWindow.BODY_INPUT;
+      bodyInput.placeholder = MarkerInfoWindow.DEFAULT_BODY;
+
+      // if this marker not new, populate the form with the current information
+      if (this.myMarker.isEditing()) {
+        if (this.myMarker.title) {
+          titleInput.innerHTML = this.myMarker.title;
+        }
+        if (this.myMarker.body) {
+          bodyInput.innerHTML = this.myMarker.body;
+        }
+      }
+
+      let submitBtn = document.createElement("button");
+      submitBtn.id = MarkerInfoWindow.SUBMIT_BTN;
+      submitBtn.innerHTML = "Enter";
+
+      form.appendChild(titleInput);
+      form.appendChild(bodyInput);
+      form.appendChild(submitBtn);
+      return form;
+    }
+
+    /**
+     * Puts together the title and body of the info window
+     */
+    makeRightColumn() {
+      let rightCol = document.createElement("div");
+      rightCol.classList.add(MarkerInfoWindow.RIGHT_COLUMN);
+      if (this.myMarker.isEditing()) {
+        rightCol.appendChild(this.makeContentForm());
+      } else {
+        rightCol.appendChild(this.makeTitle());
+        rightCol.appendChild(this.makeBody());
+      }
+      return rightCol;
+    }
+
+    /**
+     * Returns the html string to be displayed in this info window
+     */
+    makeContent() {
```

Comment:
In the Google JS style, private methods must have a trailing '_' and have a line that says "@private" in JsDoc. Please update your private methods to follow this style. This would ensure that you and/or other developers don't accidentally call them from outside the class

Reply:
Will do, I'll also add getters and make some variables private while I'm at it.